pphysx3d/src/shapes/mod.rs

```diff
@@ -23,7 +26,62 @@ struct Ray {
 pub struct GameObject {
     pub shape: Box<dyn Shape>,    // The collider
     pub position: Isometry3<f32>, // includes a translation vector and a rotation part as an unit quaternion
-    pub velocity: Vector3<f32>,
+    velocity: Vector3<f32>,
+    acceleration: Vector3<f32>,
+    force_accum: Vector3<f32>,
     //texture:
-    pub mass: f32,
+    pub inverse_mass: f32,
+    pub restitution: f32, // elasticity aka bounciness. rename to bounciness?
+    pub friction: f32,    // coefficient of friction
+}
+
+impl GameObject {
+    pub fn set_velocity(&mut self, velocity: Vector3<f32>) {
+        self.velocity = velocity;
+    }
+
+    pub fn get_velocity(&self) -> Vector3<f32> {
```

Comment:
this is not rust idiomatic. if a field is gettable and settable it should just be public

Reply:
Ah alright, wasn't sure. Was switching back and forth trying to get all mutable borrows to work correctly. Will turn it into public field instead.